File: 2017/02/main.go
package main

import (
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"
    "math"
)

func loadInput() string {
    content, err := ioutil.ReadFile("input.txt")
    if err != nil {
        fmt.Println("Error reading file:", err)
    }
    str := string(content)
    return str
}

func partOne() int {
    str := strings.Split(loadInput(), "\n")
    sum := 0
    for i := 0; i < len(str); i++ {
        min_val := math.MaxInt64
        max_val := math.MinInt64
        numbers := strings.Split(str[i], "\t")
        for j := 0; j<len(numbers); j++{
        num, err := strconv.Atoi(string(numbers[j]))
        if err != nil {
        fmt.Println("Error converting string:", err)
    }
        if num > max_val{
        max_val = num
        }
        if num < min_val{
        min_val = num
        }
        }
        sum += max_val - min_val
}
    return sum
}

func partTwo() int {
    str := strings.Split(loadInput(), "\n")
    sum := 0
    for i := 0; i < len(str); i++ {
        numbers := strings.Split(str[i], "\t")
        for j := 0; j<len(numbers); j++{
        for h := 0; h<len(numbers); h++{
        num, err := strconv.Atoi(string(numbers[j]))
        if err != nil {
        fmt.Println("Error converting string:", err)
    }
        number, err := strconv.Atoi(string(numbers[h]))
        if err != nil {
        fmt.Println("Error converting string:", err)
    }

        if num != number && num % number==0{
        sum += num/number
        }
        }
        }

}
    return sum
}

func main() {
    num := partOne()
    fmt.Println(num)
    number := partTwo()
    fmt.Println(number)
}

File: 2017/02/main.js
const fs = require('fs');
function load_input(filename = 'input.txt'){
 try {
     return fs.readFileSync('input.txt', 'utf8')
    }
catch (err) {
  console.error(err);
    }
}
const input_data = load_input()


function part_one(input_data) {
    checksum = 0
    let lines = input_data.split("\n")
    for (let i= 0; i < lines.length; i++) {
        numbers = lines[i].split("\t")
        numbers = numbers.map(function(str) {
    return parseInt(str, 10);
});
        checksum += Math.max(...numbers)- Math.min(...numbers)
    }
    return checksum
}

function part_two(input_data) {
    checksum = 0
    let lines = input_data.split("\n")
    for (let i= 0; i < lines.length; i++) {
        numbers = lines[i].split("\t")
        numbers = numbers.map(function (str) {
            return parseInt(str, 10);
        })
        for (let j = 0; j < numbers.length; j++) {
            for (let h = 0; h < numbers.length; h++) {
                if (numbers[j] % numbers [h] === 0 && numbers[j] !== numbers[h]) {
                    checksum += numbers[j] / numbers [h]
                }

            }
        }
    }
    return checksum
}

console.log("---PART ONE---")
console.log(part_one(input_data))
console.log("---PART TWO---")
console.log(part_two(input_data))
